UI/src/view/MassTransitStopForm.js

```diff
@@ -547,13 +658,28 @@
       eventbus.on('assetPropertyValue:changed', function (event) {
         var property = event.propertyData;
 
-        if(event.id && property.publicId === 'tietojen_yllapitaja' && (_.find(property.values, function (value) {return value.propertyValue == '2';}))){
+        // Handle form reload after administrator change
+        if (property.publicId === 'tietojen_yllapitaja' && (_.find(property.values, function (value) {return value.propertyValue == '2';}))) {
+          isAdministratorELYKeskus = true;
+          property.propertyType = "single_choice";
+          renderAssetForm();
+        }
+        if (property.publicId === 'tietojen_yllapitaja' && (_.find(property.values, function (value) {return value.propertyValue != '2';}))) {
```

Comment:
Can be more simplified, for example modified the second IF to ELSE IF, or put the first conditions in a IF and inside evaluate the propertyValue, if is equals to 2 do some actions, ELSE do other actions.